backend/src/components/licence.js

```diff
@@ -25,6 +24,60 @@ async function getLicences(req, res) {
   }
 }
 
+async function getLicencesByFundingAgreementId(fundingAgreementId, res) {
+  try {
+    const operation =
+      'ccof_funding_agreements?$select=ccof_funding_agreementid' +
+      '&$expand=ccof_Funding_Agreement_ccof_license_ccof_license' +
+      '($select=ccof_end_date,ccof_extended_days_per_week,ccof_extended_hours_offered,ccof_extended_weeks_per_year,ccof_facility_id,ccof_maximum_days_per_week,ccof_maximum_weeks_per_year,' +
+      'ccof_name,ccof_record_end_date,ccof_record_start_date,ccof_start_date,_ccof_facility_value,ccof_licenseid,ccof_maximum_capacity,ccof_organization,statuscode;' +
+      '$filter=(statecode eq 0 and statuscode ne 100000001 and statuscode ne 1))' +
+      `&$filter=(ccof_funding_agreementid eq ${fundingAgreementId})`;
+    const response = await getOperation(operation);
+    const licences = [];
+    for (const fa of response.value) {
+      for (const l of fa.ccof_Funding_Agreement_ccof_license_ccof_license) {
+        licences.push({
+          ...new MappableObjectForFront(l, LicenceMappings).toJSON(),
+          serviceDeliveryDetails: (await getRawServiceDetails(l.ccof_licenseid)).map((d) => ({
+            ...new MappableObjectForFront(d, ServiceDeliveryMappings).toJSON(),
+            facilityName: d.ccof_license?.ccof_facility?.name,
+          })),
+        });
+      }
+    }
+    return res.status(HttpStatus.OK).json(licences);
```

Comment:
I think we're going to need to restrict licences based on facility. See the function `restrictFacilities()`.